snuba/clickhouse/native.py

```diff
@@ -81,12 +94,36 @@ def __init__(
         self.client_settings = client_settings
 
         self.pool: queue.LifoQueue[Optional[Client]] = queue.LifoQueue(max_pool_size)
+        self.fallback_pool: queue.LifoQueue[Optional[Client]] = queue.LifoQueue(
+            self.FALLBACK_POOL_SIZE
+        )
         self.__gauge = ThreadSafeGauge(metrics, "connections")
 
         # Fill the queue up so that doing get() on it will block properly
         for _ in range(max_pool_size):
             self.pool.put(None)
 
+        for _ in range(self.FALLBACK_POOL_SIZE):
+            self.fallback_pool.put(None)
+
+    def fallback_pool_enabled(self) -> bool:
+        return state.get_config("use_fallback_host_in_native_connection_pool", 0) == 1
+
+    def get_fallback_host(self) -> Tuple[str, int]:
+        config_hosts_str = state.get_config(
+            f"fallback_hosts:{self.host}:{self.port}", None
+        )
+        assert config_hosts_str, f"no fallback hosts found for {self.host}:{self.port}"
+
+        config_hosts = cast(str, config_hosts_str).split(",")
+        selected_host_port = random.choice(config_hosts).split(":")
```

Comment:
What happens if the fallback host is also the bad host?

Reply:
It'll fail the way it does today if a single attempt fails